packages/components/src/internal/components/lineage/actions.ts

```diff
@@ -284,54 +283,45 @@ export function loadSeedResult(seed: string, container?: string, options?: Linea
 }
 
 // TODO add jest test coverage for this function
-function computeSampleCounts(lineageResult: LineageResult, sampleSets: any): any {
+function computeSampleCounts(lineageResult: LineageResult, sampleSets: ISelectRowsResult): any {
     const { key, models } = sampleSets;
+    const { nodes, seed } = lineageResult;
 
-    const rows = [];
     const nodeIds = {};
+    const seedNode = nodes.get(seed);
 
-    lineageResult.nodes.forEach(node => {
-        if (node.lsid && node.cpasType) {
-            const key = node.cpasType;
-
-            if (!nodeIds[key]) {
-                nodeIds[key] = [];
+    nodes.forEach(({ cpasType, id, lsid }) => {
+        if (lsid && cpasType) {
+            if (!nodeIds[cpasType]) {
+                nodeIds[cpasType] = [];
             }
 
-            nodeIds[key].push(node.id);
+            nodeIds[cpasType].push(id);
         }
     });
 
-    for (const row in models[key]) {
-        if (models[key].hasOwnProperty(row)) {
-            const _row = models[key][row];
-
-            let count = 0,
-                filteredURL;
-            const name = _row['Name'].value,
-                ids = nodeIds[_row['LSID'].value];
+    const rows = Object.values(models[key]).map(row => {
+        const name = caseInsensitive(row, 'Name').value;
+        const cpasType = caseInsensitive(row, 'LSID').value;
+        const count = nodeIds[cpasType]?.length ?? 0;
 
-            // if there were related samples, use the array of RowIds as a count and to build an AppURL and filter
-            if (ids) {
-                count = ids.length;
-
-                filteredURL = AppURL.create('samples', name)
-                    .addFilters(Filter.create('RowId', ids, Filter.Types.IN))
-                    .toHref();
-            }
+        return {
+            name: {
+                url: AppURL.create('samples', name).toHref(),
+                value: name,
+            },
+            sampleCount: {
+                url:
+                    count > 0
+                        ? AppURL.create('rd', 'expdata', seedNode.id, 'samples').addParam('tab', name).toHref()
```

Comment:
minor: use `SAMPLES_KEY` const

Reply:
This isn't actually the same thing. This is a new route that exists in LKB's registry to a new tab called "Samples" which displays all derived samples of a registry item.

Reply:
Ah ok, got it. That makes sense.